src/main/java/org/jabref/logic/importer/fileformat/BiblioscapeImporter.java

```diff
@@ -255,7 +256,7 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
                 if (!comments.isEmpty()) { // set comment if present
                     hm.put(FieldName.COMMENT, String.join(";", comments));
                 }
-                BibEntry b = new BibEntry(bibtexType);
+                BibEntry b = new BibEntry(BibtexEntryTypes.getType(bibtexType).get());
```

Comment:
This may throw a NPE, because `BibtexEntryTypes.getType(bibtexType)` may return an empty optional. To handle such cases, one would need to add code similar to how you did it in BibtexParser. In order to reuse this code, I would propose to add a `getTypeOrDefault` method to `BibtexEntryTypes`.

Reply:
True

Reply:
Thanks. There are still a few other places where the new method should be used, this is also the reason why some of the unit tests fail at the moment.

Reply:
You're right. Fixed several tests all related with this NPE problem but some are failing. I switch to master and noticed that these are also failing on master so I believe these are non related.